pegasus/sites.v3/code.org/public/js/volunteer-local.js

```diff
@@ -153,194 +165,239 @@ function getLocations(results) {
+  }
+
+  return elem;
+}
+
 function compileHTML(index, location) {
-  var lines = [];
-  var line;
+  let elements = [];
 
-  // Compile HTML.
-  var html = '<h3 class="entry-detail">' + location.name_s + '</h3>';
+  elements.push(
+    createHTMLElement("h3", { class: "entry-detail" }, location.name_s)
+  );
 
   if (location.company_s) {
-    line = location.company_s;
-    lines.push(line);
+    elements.push(document.createTextNode(location.company_s));
   }
 
   if (location.experience_s) {
-    line = '<strong>Experience: </strong>' + i18n(location.experience_s);
-    lines.push(line);
+    elements.push(createHTMLElement("strong", null, "Experience:"));
+    elements.push(document.createTextNode(location.experience_s));
   }
 
   if (location.location_flexibility_ss) {
-    $.each(location.location_flexibility_ss, function (key, field) {
-      location.location_flexibility_ss[key] = i18n('location_' + field);
+    $.each(location.location_flexibility_ss, function(key, field) {
+      location.location_flexibility_ss[key] = i18n("location_" + field);
     });
 
-    line = '<strong>How I can help: </strong>' + location.location_flexibility_ss.join(', ');
-    lines.push(line);
+    elements.push(createHTMLElement("strong", null, "How I can help:"));
+    elements.push(
+      document.createTextNode(location.location_flexibility_ss.join(", "))
+    );
   }
 
   if (location.description_s) {
-    line = '<strong>About me: </strong>' + location.description_s;
-    lines.push(line);
+    elements.push(createHTMLElement("strong", null, "About me:"));
+    elements.push(document.createTextNode(location.description_s));
   }
 
   if (location.linkedin_s) {
     if (!location.linkedin_s.match(/^https?:\/\//i)) {
-      location.linkedin_s = 'http://' + location.linkedin_s;
+      location.linkedin_s = "http://" + location.linkedin_s;
     }
 
-    line = '<a href="' + location.linkedin_s + '" target="_blank">LinkedIn profile</a>';
-    lines.push(line);
+    elements.push(createHTMLElement("strong", null, "LinkedIn profile:"));
+    elements.push(document.createTextNode(location.linkedin_s));
   }
 
   if (location.facebook_s) {
     if (!location.facebook_s.match(/^https?:\/\//i)) {
-      location.facebook_s = 'http://' + location.facebook_s;
+      location.facebook_s = "http://" + location.facebook_s;
     }
 
-    line = '<a href="' + location.facebook_s + '" target="_blank">Facebook profile</a>';
-    lines.push(line);
+    elements.push(createHTMLElement("strong", null, "Facebook profile:"));
+    elements.push(document.createTextNode(location.facebook_s));
   }
 
-  $.each(lines, function (key, field) {
-    html += '<div class="profile-detail entry-detail">' + field + '</div>';
+  let htmlStr = "";
+  elements.forEach(elem => {
+    let div = createHTMLElement("div", {
+      class: "profile-detail entry-detail"
+    });
+    div.appendChild(elem);
+    htmlStr += div.outerHTML;
   });
```

Comment:
I believe your change adds up to five additional lines to the profile bubble.

Here's how it looks now, with the `strong` headings inline with content:
![image](https://user-images.githubusercontent.com/1615761/65376739-73d3bd80-dc58-11e9-885b-6c90c82ade46.png)

Two of them (the LinkedIn and Facebook lines) seem fine since we're changing them anyway.  But in other places like "About me" what used to be one push onto `lines` has become two pushes onto `elements`:

```diff
-    line = '<strong>About me: </strong>' + location.description_s;
-    lines.push(line);
+    elements.push(createHTMLElement("strong", null, "About me:"));
+    elements.push(document.createTextNode(location.description_s));
```

It'd be nice to stick a little closer to current formatting, since these bubbles often hit their max-height and require scrolling as it is:

![image](https://user-images.githubusercontent.com/1615761/65376810-5b17d780-dc59-11e9-9508-8d6edf9733e6.png)

Reply:
Will fix it to keep the original format.